Source/WTF/wtf/SortedArrayMap.h

```diff
@@ -230,51 +228,47 @@ template<typename ArrayType> template<typename KeyArgument> inline bool SortedAr
     return iterator != std::end(m_array) && *iterator == *parsedKey;
 }
 
-constexpr int strcmpConstExpr(const char* a, const char* b)
+constexpr int compareSpansConstExpr(std::span<const char> a, std::span<const char> b)
 {
-    while (*a == *b && *a && *b) {
-        ++a;
-        ++b;
-    }
-    return *a == *b ? 0 : *a < *b ? -1 : 1;
+    auto commonLength = std::min(a.size(), b.size());
+    size_t i = 0;
+    while (i < commonLength && a[i] == b[i])
+        ++i;
+    auto aCharacter = i < a.size() ? a[i] : 0;
+    auto bCharacter = i < b.size() ? b[i] : 0;
```

Comment:
This does not seem quite right. We don't want lack of a character to sort equal to a 0 character. The lack of a character should sort as less than any character. It was different when the strings were null-terminated.

Reply:
I was trying to match the original code. The original code was dropping out of the loop if we reached a character mismatch and or a null terminator.

Falling back to 0 here when `I` is larger then than span size was my attempt to keep the same behavior as with nun-terminated strings. The previous code was using 0 in the comparison if the loop above aborted due to reaching the null terminator.

Reply:
Yes I understand that you were preserving behavior. My code was wrong but what we will want here is something that treats a short string as less than a longer one with the same prefix with no special handling of null characters.